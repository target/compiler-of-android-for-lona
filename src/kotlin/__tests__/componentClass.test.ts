import { createComponentClass } from '../componentClass'

describe('Kotlin / Component', () => {
  it('creates a component class', () => {
    const componentClass = createComponentClass({
      namePrefix: 'Example',
      packagePath: 'com.example',
      imports: ['android.widget.TextView'],
      parameters: [
        {
          name: 'titleText',
          type: 'CharSequence',
          defaultValue: `""`,
          styleableAttribute: {
            getter: 'getString',
            format: 'string',
          },
        },
      ],
      publicViews: [
        {
          name: 'text',
          type: 'TextView',
        },
      ],
    })

    expect(componentClass).toMatchSnapshot()
  })
})
