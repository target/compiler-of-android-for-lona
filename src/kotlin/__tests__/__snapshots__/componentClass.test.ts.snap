// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kotlin / Component creates a component class 1`] = `
"package com.example

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import androidx.constraintlayout.widget.ConstraintLayout
import com.example.databinding.ExampleBinding
import android.widget.TextView

open class ExampleView : ConstraintLayout {

    //region Public

    open var titleText: CharSequence = \\"\\"
      set(value) {
        field = value
        update()
      }
    
    open val textView: TextView by lazy { binding.text }

    //endregion
  
    //region Private

    private lateinit var binding: ExampleBinding

    //endregion

    //region Lifecycle

    constructor(context: Context) : super(context) {
        initialize(context, null)
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        initialize(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        initialize(context, attrs)
    }

    private fun initialize(context: Context, attrs: AttributeSet?) {
        binding = ExampleBinding.inflate(LayoutInflater.from(context), this, true)

        attrs?.let {
            val typedAttributes = context.obtainStyledAttributes(it, R.styleable.ExampleView)

            titleText = typedAttributes.getString(R.styleable.ExampleView_title_text) ?: titleText
        }
    }

    private fun update() {}

    //endregion
}
"
`;
